WIP docs enhancements (#450)
src-d/lookout: docs/configuration.md

# Configuring source{d} Lookout

> The behavior of **source{d} Lookout** is configured with two different configuration files:

Reviewer: Too many _configure_
```suggestion
The behavior of **source{d} Lookout** is defined with two different configuration files:
```

> - [`config.yml`](#config.yml), to define the global configuration of the server.

Reviewer: Link does not work, at least in GitHub, I don't know about Gitbook.
```suggestion
- [`config.yml`](#configyml), to define the global configuration of the server.
```

Author: I asumed GitBook as the preferred way to render our docs, and it works differently when creating anchors for titles having `.` inside.
- Title: `config.yml`
- GitBook: `#config-yml`
- GitHub: `#configyml`

In case it is required to support both, would you propose an alternative?

Reviewer: For this PR it would be fine if it only works on GitBook, in my opinion. But if the section links do not work on GitHub let's at least open an issue and try to fix it at some point. Maybe we just need to assume the limitations and avoid `.` in titles.

Author: agree, issue created https://github.com/src-d/lookout/issues/469

> - [`.lookout.yml`](#.lookout.yml), to override the analyzer default behavior in each repository.

Reviewer: Same,
```suggestion
- [`.lookout.yml`](#lookoutyml), to override the analyzer default behavior in each repository.
```

# config.yml

> **source{d} Lookout** is configured with the `config.yml` file, you can use the template [`config.yml.tpl`](/config.yml.tpl) to create your own. Use the `lookoutd` option `--config` to set the path to it, or use the default location at `./config.yml`. The config file is read on server startup, so it is needed to be restarted to load a new configuration.

Reviewer: ```suggestion
**source{d} Lookout** is configured with the `config.yml` file, you can use the template [`config.yml.tpl`](/config.yml.tpl) to create your own. Use the `lookoutd` option `--config` to set the path to it, or use the default location at `./config.yml`. The config file is read on server startup, so `lookoutd` needs to be restarted in order to load a new configuration.
```

The most important things you need to configure for a local installation, are:

> 1. [Repositories](#repositories): add the URLs of the repositories to be watched.

Reviewer: This is only the case if you are using GitHub token instead of GitHub App auth. This is getting a bit confusing, and I think we should consider separating both auth methods more clearly in the docs. If that case maybe it makes sense to leave this line as you propose, and then come back to it in a later PR.

Author: What if reworded this other way?
```suggestion
1. [Repositories](#repositories): define the repositories to be watched.
```
This way we only link to `Repositories` section, where it's explained the two different ways to specify the repositories to be watched.

Reviewer: :+1:
